### PR TITLE
[TLS] Update DEK usage according to dpcp 1.1.44

### PR DESCRIPTION
**Pull request by galnoam**
RM: 3705189

DESC: DEK API was changed in dpcp version 1.1.44
      Need to update usage of dpcp::dek

### CODE REVIEW

**Comment by galnoam** [comment]
@pasis please don't merge it, it was not tested properly yet.

File: src/core/dev/qp_mgr_eth_mlx5.cpp [hunk 1]
```diff
@@ -761,19 +761,19 @@ int qp_mgr_eth_mlx5::send_to_wire(xlio_ibv_send_wr *p_send_wqe, xlio_wr_tx_packe
 
 #ifdef DEFINED_UTLS
 
-std::unique_ptr<dpcp::dek> qp_mgr_eth_mlx5::get_new_dek(const void *key, uint32_t key_size_bytes)
+std::unique_ptr<dpcp::tls_dek> qp_mgr_eth_mlx5::get_new_tls_dek(const void *key, uint32_t key_size_bytes)
```
Comment (alexbriskin) [inline src/core/dev/qp_mgr_eth_mlx5.cpp:764]:
```suggestion
std::unique_ptr<dpcp::tls_dek> qp_mgr_eth_mlx5::get_new_tls_dek(const void *key,
                                                                uint32_t key_size_bytes)
```

File: src/core/dev/qp_mgr_eth_mlx5.cpp [hunk 2]
```diff
@@ -783,16 +783,16 @@ std::unique_ptr<dpcp::dek> qp_mgr_eth_mlx5::get_new_dek(const void *key, uint32_
         }
     }
 
-    return std::unique_ptr<dpcp::dek>(_dek);
+    return std::unique_ptr<dpcp::tls_dek>(_dek);
 }
 
-std::unique_ptr<dpcp::dek> qp_mgr_eth_mlx5::get_dek(const void *key, uint32_t key_size_bytes)
+std::unique_ptr<dpcp::tls_dek> qp_mgr_eth_mlx5::get_tls_dek(const void *key, uint32_t key_size_bytes)
```
Comment (alexbriskin) [inline src/core/dev/qp_mgr_eth_mlx5.cpp:789]:
```suggestion
std::unique_ptr<dpcp::tls_dek> qp_mgr_eth_mlx5::get_tls_dek(const void *key,
                                                            uint32_t key_size_bytes)
```

File: src/core/dev/qp_mgr_eth_mlx5.cpp [hunk 3]
```diff
@@ -1336,7 +1336,10 @@ void qp_mgr_eth_mlx5::ti_released(xlio_ti *ti)
 
 void qp_mgr_eth_mlx5::put_tls_tis_in_cache(xlio_tis *tis)
 {
-    put_dek(tis->release_dek());
+    std::unique_ptr<dpcp::dek> dek = tis->release_dek();
+    assert(dynamic_cast<dpcp::tls_dek*>(dek.get()) != nullptr);
```
Comment (alexbriskin) [inline src/core/dev/qp_mgr_eth_mlx5.cpp:1340]:
```suggestion
    assert(dynamic_cast<dpcp::tls_dek *>(dek.get()) != nullptr);
```

File: src/core/dev/qp_mgr_eth_mlx5.cpp [hunk 4]
```diff
@@ -1336,7 +1336,10 @@ void qp_mgr_eth_mlx5::ti_released(xlio_ti *ti)
...
+
+    put_tls_dek(std::unique_ptr<dpcp::tls_dek>(dynamic_cast<dpcp::tls_dek*>(dek.release())));
```
Comment (alexbriskin) [inline src/core/dev/qp_mgr_eth_mlx5.cpp:1342]:
```suggestion
    put_tls_dek(std::unique_ptr<dpcp::tls_dek>(dynamic_cast<dpcp::tls_dek *>(dek.release())));
```

**Review by pasis — APPROVED** [review]

**Pull request merged**